src/simplewallet/simplewallet.cpp

```diff
@@ -362,6 +362,33 @@ bool simple_wallet::seed_set_language(const std::vector<std::string> &args/* = s
   return true;
 }
 
+bool simple_wallet::change_password(const std::vector<std::string> &args)
+{ 
+  const auto orig_pwd_container = get_and_verify_password();
+
+  if(orig_pwd_container == boost::none)
+  {
+    fail_msg_writer() << tr("Your original password was incorrect.");
+    return false;
+  }
+
+  // prompts for a new password, this is not a new wallet so pass in false.
+  const auto pwd_container = tools::wallet2::password_prompt(false);
+  
+  try
+  {
+    m_wallet->rewrite(m_wallet_file, pwd_container->password());
+    m_wallet->store();
+  }
+  catch (const wallet_logic_error& e)
```

Comment:
Not certain what the correct type is.  Replacing with std::logic_error resolves this problem during compile.
/home/tdprime/bitmonero/src/simplewallet/simplewallet.cpp:383:16: error: ISO C++ forbids declaration of ‘wallet_logic_error’ with no type [-fpermissive]
   catch (const wallet_logic_error& e)
                ^
/home/tdprime/bitmonero/src/simplewallet/simplewallet.cpp:383:34: error: expected ‘)’ before ‘&’ token
   catch (const wallet_logic_error& e)
                                  ^
/home/tdprime/bitmonero/src/simplewallet/simplewallet.cpp:383:34: error: expected ‘{’ before ‘&’ token
/home/tdprime/bitmonero/src/simplewallet/simplewallet.cpp:383:36: error: ‘e’ was not declared in this scope
   catch (const wallet_logic_error& e)
                                    ^